config/application_space_test.go

```diff
@@ -0,0 +1,321 @@
+			).Return(beResp, nil)
+
+			resp, err := configClient.CreateApplicationSpace(ctx, req)
+			Expect(err).To(Succeed())
+			Expect(resp).To(test.EqualProto(beResp))
+		})
+	})
+	Describe("AppSpaceUpdate", func() {
+		DescribeTable("Request error during Update ",
+			func(req *configpb.UpdateApplicationSpaceRequest, causeMatcher, msgMatcher, codeMatcher OmegaMatcher) {
+				resp, err := configClient.UpdateApplicationSpace(ctx, req)
+				Expect(err).To(HaveOccurred())
+				Expect(resp).To(BeNil())
+
+				var clientErr *sdkerrors.ClientError
+				Expect(errors.As(err, &clientErr)).To(BeTrue(), "is client error")
+				Expect(clientErr.Unwrap()).To(causeMatcher)
+				Expect(clientErr.Message()).To(msgMatcher)
+				Expect(clientErr.Code()).To(codeMatcher)
+			},
+			Entry("nil request", nil, BeNil(), Equal("invalid nil request"), Equal(codes.InvalidArgument)),
+		)
+
+		It("Update", func() {
+			displayNamePb := &wrapperspb.StringValue{Value: "Like real AppSpace Name"}
+			etagPb := &wrapperspb.StringValue{Value: "123qwert"}
+			req := &configpb.UpdateApplicationSpaceRequest{
+				Id:          "gid:like-real-app-space-id",
+				Etag:        etagPb,
+				DisplayName: displayNamePb,
+			}
+			beResp := &configpb.UpdateApplicationSpaceResponse{
+				Id:   "gid:like-real-app_space-id",
+				Etag: "123qwert",
+			}
+
+			mockClient.EXPECT().UpdateApplicationSpace(
+				gomock.Any(),
+				test.WrapMatcher(test.EqualProto(req)),
+				gomock.Any(),
+			).Return(beResp, nil)
+
+			resp, err := configClient.UpdateApplicationSpace(ctx, req)
+			Expect(err).To(Succeed())
+			Expect(resp).To(test.EqualProto(beResp))
+		})
+	})
+	Describe("AppSpaceList", func() {
+		DescribeTable("Request error during List ",
+			func(req *configpb.ListApplicationSpacesRequest, causeMatcher, msgMatcher, codeMatcher OmegaMatcher) {
+				resp, err := configClient.ListApplicationSpaces(ctx, req)
+				Expect(err).To(HaveOccurred())
+				Expect(resp).To(BeNil())
+
+				var clientErr *sdkerrors.ClientError
+				Expect(errors.As(err, &clientErr)).To(BeTrue(), "is client error")
+				Expect(clientErr.Unwrap()).To(causeMatcher)
+				Expect(clientErr.Message()).To(msgMatcher)
+				Expect(clientErr.Code()).To(codeMatcher)
+			},
+			Entry("nil request", nil, BeNil(), Equal("invalid nil request"), Equal(codes.InvalidArgument)),
+		)
+
+		It("MatchName", func() {
+			mockResponseClient := configmock.NewMockConfigManagementAPI_ListApplicationSpacesClient(mockCtrl)
+			mockListApplicationSpacesRequest := configpb.ListApplicationSpacesRequest{
+				CustomerId: "gid:like-real-customer-id",
+				Match:      []string{"like-real-appspace-name"},
+			}
+
+			mockClient.EXPECT().ListApplicationSpaces(
+				gomock.Any(),
+				gomock.Eq(&mockListApplicationSpacesRequest),
+				gomock.Any(),
+			).Return(mockResponseClient, nil)
+
+			resp, err := configClient.ListApplicationSpaces(context.Background(), &mockListApplicationSpacesRequest)
+			Ω(resp).ToNot(BeNil())
+			Ω(err).To(Succeed())
+		})
```

Comment:
This test doesn't test much. You should mock some responses and verify List returns them.

Reply:
I need help here: each time I change something I have error like, when I want a complete answer:  [FAILED] wrong type of argument 0 to Return for *config.MockConfigManagementAPIClient.ListApplicationSpaces: *configv1beta1.ApplicationSpace is not assignable to configv1beta1.ConfigManagementAPI_ListApplicationSpacesClient